HIP/hip-991.md

```diff
@@ -44,182 +44,310 @@ This document outlines the development of a fixed fee system for the submission
+* **Topic Operator:** Entities or individuals who manage and operate topics, interested in stable revenue and easy management of topic-related transactions.
+* **Topic Operator:** Entities or individuals who manage and operate topics, interested in reducing the noise of public topic IDs/making it more difficult to spam an open topic.
+* **Regular User:** End users who interact with topics, adding paid methods of interaction and submission to topics.
 
-### **FixedFee**
+**User Stories:**
 
-The `FixedFee` message represents a fixed fee charged for each message submission to a topic. This protobuf message already exists so it can be re-used.
+* As a **Topic Operator**, I want to set a fixed fee for submitting messages to my topic, so that I can ensure a consistent revenue stream and manage my topic effectively.
+* As a **Regular User**, I want to know the exact cost of interacting with a topic.
 
-```
-message FixedFee {
-  // The number of units to assess as a fee
-  int64 amount = 1;
-  // The denomination of the fee; taken as hbar if left unset
-  TokenID denominating_token_id = 2;
-}
-```
+## Specification
 
-### **CustomFixedFee**
+### Overview
 
-The `CustomFixedFee` message defines the type of fee and the account receiving the fee assessed during a message submission to the associated topic. A custom fee may only be fixed and must specify a fee collector account to receive the assessed fees.
+We propose adding a fixed fee mechanism to the Hedera Consensus Service (HCS) for topic messages, similar to the custom fee structures in the Hedera Token Service (HTS). This will involve creating new protocol buffers (protobuf) messages and modifying existing ones to accommodate fixed fees for topics.
 
-```
-message CustomFixedFee {
-  // Fixed fee to be charged
-  FixedFee fixed_fee = 1;
+### Requirements and Scope
 
-  // The account to receive the custom fee
-  AccountID fee_collector_account_id = 3;
-}
-```
+* Topics can have a Fee Schedule Key set at creation time.
+* Fee Schedule Key can manage fee updates and send free messages to the topic.
+* The Fee Schedule Key can be updated with an update topic transaction signed by the Admin Key and the new Fee Schedule Key.
+* If the topic was created without a Fee Schedule Key, the key cannot be added later.
+* Topics can have fees for submitting messages.
+* Fees can be set at creation time.
+* Fees can be changed (updated or removed) from a topic with an update topic transaction signed by the Fee Schedule Key.
+* Fees are defined as a list of custom fee.
+* A custom fee can be set in HBAR or HTS fungible tokens, and must have an accountID as a receiver.
+* Fee payer must set an allowance in HBAR or HTS tokens to be used to pay fees for message submissions, on an HCS topic basis.
+* Allowance granted to topic ID does not allow the topic's Admin Key (or any of the topic's keys) to spend that allowance.
+* Fee payer must set can set a maximum fee per message, on an HCS topic basis.
```

Comment:
How does this interact with allowance / is it a max per message allowance, or a total allowance, and a max fee per message?

Reply:
See above, the suggestion is to have both options available.